pulsar-client/src/main/java/org/apache/pulsar/client/impl/TableViewImpl.java

```diff
@@ -224,9 +224,23 @@ private CompletableFuture<Reader<T>> newReader(String partition) {
                 .readCompacted(true)
                 .poolMessages(true)
                 .createAsync()
+                .thenCompose(this::cacheNewReader)
                 .thenCompose(this::readAllExistingMessages);
     }
 
+    private CompletableFuture<Reader<T>> cacheNewReader(Reader<T> reader) {
+        CompletableFuture<Reader<T>> future = new CompletableFuture<>();
+        if (this.readers.containsKey(reader.getTopic())) {
```

Comment:
Why  `readers.containsKey` checked twice?  the previous check is in `start` method
```
 // Add new Partitions
                    partitions.forEach(partition -> {
                        if (!readers.containsKey(partition)) {
                            futures.add(newReader(partition));
                        }
                    });
```

Reply:
This is safe-guard condition to make the future finish in both cases and for code completeness